mne/decoding/transformer.py

```diff
@@ -715,3 +719,166 @@ def transform(self, X):
+        See mne.filter.construct_iir_filter for details. If iir_params
+        is None and method="iir", 4th order Butterworth will be used.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+        Defaults to self.verbose.
+
+    See Also
+    --------
+    FilterEstimator
+    Vectorizer
+    """
+    def __init__(self, l_freq=None, h_freq=None, sfreq=1.0,
+                 filter_length='auto', l_trans_bandwidth='auto',
+                 h_trans_bandwidth='auto', n_jobs=1, method='fir',
+                 iir_params=None, verbose=None):
+        self.l_freq = l_freq
+        self.h_freq = h_freq
+        self.filter_length = filter_length
+        self.l_trans_bandwidth = l_trans_bandwidth
+        self.h_trans_bandwidth = h_trans_bandwidth
+        self.n_jobs = n_jobs
+        self.method = method
+        self.iir_params = iir_params
+        self.sfreq = sfreq
+
+        dummy_array = [1, 2]
+
+        (_, self.sfreq, self.l_freq, self.h_freq, self.l_trans_bandwidth,
+         self.h_trans_bandwidth, self.filter_length, _) = \
+            _triage_filter_params(dummy_array, sfreq, l_freq, h_freq,
+                                  l_trans_bandwidth, h_trans_bandwidth,
+                                  filter_length, method, phase='zero')
+
+        if not isinstance(self.n_jobs, int) and self.n_jobs == 'cuda':
+            raise ValueError('n_jobs must be int or "cuda", got %s instead.'
+                             % type(self.n_jobs))
+
+        if self.l_freq == 0:
+            self.l_freq = None
+        if self.h_freq is not None and self.h_freq > (self.sfreq / 2.):
+            self.h_freq = None
+        if self.l_freq is not None and not isinstance(self.l_freq, float):
+            self.l_freq = float(self.l_freq)
+        if self.h_freq is not None and not isinstance(self.h_freq, float):
+            self.h_freq = float(self.h_freq)
+
+    def fit(self, X, y=None):
+        """Does nothing. For scikit-learn compatibility purposes.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data.
+        y : array, shape (n_epochs,) | None
+            The label for each epoch.
+
+        Returns
+        -------
+        self : instance of Filterer
+            Returns the modified instance.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("X should be of type ndarray (got %s)."
+                             % type(X))
+
+        return self
+
+    def transform(self, X, y=None):
+        """Filters data along the last dimension.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to be filtered. Can be, for example a list, or an array
+            of at least 2d. The first dimension must be of length n_samples,
+            where samples are the independent samples used by the filter
+            (e.g. n_epochs for epoched data).
```

Comment:
This wording is confusing. In the context of MNE, samples refer to MEG/EEG samples. Can't you just say filtering is performed over the last dimension, which should therefore denote how the signal is sampled over time?